float64 x
float64 y
float64 z
---
float64 distance
float64 gx
float64 gy
float64 gz
